lib/DBusObjectLibs.js

```diff
@@ -111,60 +111,196 @@ class DBusObject {
-	let pathComponents = relativePath.split ('/')
+		// Check that 'object' is a DBusObject
+		if (! (object instanceof DBusObject)) {
+			throw new TypeError (`'object' is not a DBusObject.`)
+		}
+
+		// Check that all paths components are valid
+		if (!pathComponents.every (utils.isValidPathComponent)) {
+			throw new TypeError (`'${relativePath}' contains non-valid path components.`)
+		}
+
+		/*
+		 * Everything looks good, traverse the object according to the path components, and add the obj as child
+		 */
+		let currObj = this 
+		// traverse the object
+		while (pathComponents.length > 1) {
+			let currPathComponent = pathComponents.shift()
+
+			// If the current object doesn't already have an object at this path component, create one
+			if (typeof currObj[currPathComponent] === 'undefined') {
+				currObj[currPathComponent] = new DBusObject()
+				currObj[currPathComponent]._linkParentObject(currObj, currPathComponent)
+			}
+
+			// traverse the object
+			currObj = currObj[currPathComponent]
+		}
+
+		// Now we have traversed our object and reached the object path to host the child, so add it
+		if (currObj[pathComponents[0]]) {
+			throw new Error(`path ${relativePath} already exists`)
+		}
+		currObj[pathComponents[0]] = object
+
+		currObj[pathComponents[0]]._linkParentObject(currObj, pathComponents[0])
 
-	// Check that 'object' is a DBusObject
-	if (! (object instanceof DBusObject)) {
-		throw new TypeError (`'object' is not a DBusObject.`)
+		if (this.getService()) {
+			debug(this.getPath())
+			this.getService().bus.exposeObject(this.getService(), object, `${this.getPath()}/${relativePath}`)
+		}
 	}
 
-	// Check that all paths components are valid
-	if (!pathComponents.every (utils.isValidPathComponent)) {
-		throw new TypeError (`'${relativePath}' contains non-valid path components.`)
+	/**
+	 * Remove a given object_path subtree
+	 * @param {string} objPathComponent valid DBus object path
+	 * @throws {TypeError} if objPathComponent is not a valid dbus object path string
+	 */
+	removeObject (objPathComponent = mandatory()) {
+		if (this[objPathComponent] instanceof DBusObject) obj = this[objPathComponent]
+		else throw new TypeError(`no object at ${objPathComponent}`) 
+
+		let inputObjQueue = [obj]
+		let outputObjQueue = []
+		while (objQueue.length) {
+			let obj = objQueue.pop()
+			inputObjQueue.push.apply(inputObjQueue, obj.getChildren())
+			obj.removeAllInterfaces()
+		}
+
+		outputObjQueue.forEach(obj => obj._unlinkParentObject())
+	
+		delete this[objPathComponent]
 	}
 
-	/*
-		Everything looks good, traverse the object according to the path components, and add the obj as child
-	*/
-	let currObj = parent
-	// traverse the object
-	while (pathComponents.length > 1) {
-		let currPathComponent = pathComponents.shift()
+	/**
+	 * Get all direct sub path components of this object's path
+	 */
+	getChildrenPaths () {
+		return Object.keys(this).filter(key => this[key] instanceof DBusObject && key !== '_parentObject')
```

Comment:
If you mean DBus property, then it should not collide since the ```_parentObject``` property is a property of ```DBusObject```, not ```DBusInterface```.